### ISSUE TITLE
v14.1 Fileserver SambaDav troubles with accessing root share

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#627]
As reported in the [forums](https://www.turnkeylinux.org/forum/support/20160504/sambadav-not-working-ver-14) SambaDav in v14.1 Fileserver app is having some issues with access to the share root.

I haven't investigated at all yet, nor even tried to reproduce.

**Comment by qrntz** [comment]
The [WebDAV CGI](https://github.com/turnkeylinux/common/pull/65) migration effort should fix this along with other issues we faced with SambaDAV. Thus I don't think there is much merit in fixing this particular issue (if we patch before v14.2, might just as well go the full way with WebDAV CGI).

Speaking of that, though, the original PR became slightly outdated since master changed… let me know when you are looking to merge so I can update the branch.

**Comment by JedMeister** [comment]
I recall that one of the recent Samba bugfixes caused a regression in accessing share roots; I thought it may have been related to that?

As for your PR, apologies on the delay applying that. I have been meaning to do some testing on my end before merging but keep getting sidetracked with other stuff. I'll put it on my todo list and get back to you. Assuming that it's nothing major it's probably just as easy for me to take care of it when I do a test build.

**Comment by qrntz** [comment]
Seems like it. Honestly speaking, SambaDAV is meant primarily for use with Samba 3 so the longer Samba 4 development goes on the more actual behavior diverges from the assumptions in the code.

I'll give my branch a re-test then (just in case) and rebase it to merge cleanly.

**Comment by qrntz** [comment]
Yep, seems to work fine.

**Comment by tekbit** [comment]
I might have figured this user's question out. If you install 'smbclient' it should get resolved.

The SambaDAV relies on the 'smbclient -L' function to discover the available shared directories. Check for reference: /var/www/sambadav/config/config.inc.php and look for 'smbclient_path'. However, the 'fileserver v14.4' appliance does not come with it pre-installed.

**Comment by JedMeister** [comment]
Hmm, could be a thought. Thanks for the suggestion @tekbit

**Comment by vondrt4** [comment]
I wonder why that is. Because smbclient is included explicitly in the package install Plan `common/plans/turnkey/samba-dav`
I have observed that it gets removed during the first package upgrade operation. It is not marked Automatic in dpkg.
Running `/etc/cron-apt/action.d/5-install`:
`apt-get dist-upgrade -o APT::Get::Show-Upgraded=true -o Dir::Etc::sourcelist=/etc/apt/sources.list.d/security.sources.list -o Dir::Etc::sourceparts=nonexistent -o DPkg::Options::=--force-confdef -o DPkg::Options::=--force-confold`
...
```
The following packages will be REMOVED:
  smbclient
The following packages will be upgraded:
  libexpat1 libgd3 libidn11 libldb1 libpam-smbpass libsmbclient libssl1.0.0
  libtalloc2 libtasn1-6 libtdb1 libtevent0 libwbclient0 openssh-client
  openssh-server openssh-sftp-server openssl php-pear php5-cli php5-common
  php5-fpm python-ldb python-samba python-talloc python-tdb samba samba-common
  samba-common-bin samba-dsdb-modules samba-libs ssh tdb-tools
```
Running the 'U' action in aptitude, smbclient gets updated, not removed.

**Comment by JedMeister** [comment]
Thanks for looking into it Tomas. That is weird. I guess it's something to do with the upgrade of Samba to 4.2? Still strikes me as weird though...

So the re-installation of smbclient seems to solve the issue?!

**Comment by vondrt4** [comment]
At least I see the 3 default shares and can access the Readme file in /store. If that is a file in A share root, then it is fixed. Or was the bug about THE /root share?

**Comment by JedMeister** [comment]
I'm pretty sure that the bug reported related to the root of each share; not THE /root share. So it sounds like that resolves the issue then. Cool!

**Comment by JedMeister** [comment]
This should be resolved in v14.2. Please report if anyone is still experiencing this.
